src/Encoding/JoseEncoder.php

```diff
@@ -23,10 +24,14 @@ final class JoseEncoder implements Encoder, Decoder
     public function jsonEncode(mixed $data): string
     {
         try {
-            return json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE | JSON_THROW_ON_ERROR);
+            $jsonEncoded = json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE | JSON_THROW_ON_ERROR);
         } catch (JsonException $exception) {
             throw CannotEncodeContent::jsonIssues($exception);
         }
+
+        assert($jsonEncoded !== '');
```

Comment:
I'm not too sure whether or not the code should use a function that is affected by at least 6 configuration values in php.ini and can effectively be disabled (I believe this is even recommended for production environments)

Reply:
It's still a step up from using `/** @var non-empty-string $jsonEncoded */`.

In fact, some coding standards automatically convert this.

Adding another dependency for this is not worth it.

Reply:
And if you know how this is affecting execution: How do you document this?
- Is it throwing an exception? Which one? 
- Is it not doing anything at all? Ok.
- Warning message? Quite hard to put this into an interface annotation.
- Anything in between I have not considered? Hm...

Reply:
IMO not part of this PR's discussion: `assert()` is generally OK for type-upcasting, and well understood by PHPStan and Psalm.

As for whether they affect the runtime: doesn't really matter, since assertions of this kind are supposed to be used when things are known to be true upfront (from a developer PoV), and not as a runtime verification mechanism.

Consider an `assert()` the equivalent of an inline `@var`, but with the ability to be a bit stricter, depending on settings.

Reply:
I just updated my knowledge about assertions. :) PHP 8.0 by default throws an exception (a previously poorly documented change that was merged into PHP without too much discussion - and exactly this was triggering me).

Also I now observed this library already uses `assert()` for existing checks, so this is in line with the previous approach. 👍

Reply:
BTW, `slevomat/coding-standard` automatically converts inline `@var` into `assert()` :+1: